source/Octopus.Tentacle.Tests.Integration/ClientScriptExecutionScriptServiceV2IsNotRetriedWhenRetriesAreDisabled.cs

```diff
@@ -0,0 +1,225 @@
+            using var clientTentacle = await new ClientAndTentacleBuilder(tentacleType)
+                .WithRetriesDisabled()
+                .WithPortForwarderDataLogging()
+                .WithResponseMessageTcpKiller(out var responseMessageTcpKiller)
+                .WithRetryDuration(TimeSpan.FromMinutes(4))
+                .WithTentacleServiceDecorator(new TentacleServiceDecoratorBuilder()
+                    .LogCallsToScriptServiceV2()
+                    .RecordExceptionThrownInScriptServiceV2(out var scriptServiceExceptions)
+                    .CountCallsToScriptServiceV2(out var scriptServiceCallCounts)
+                    .DecorateScriptServiceV2With(new ScriptServiceV2DecoratorBuilder()
+                        .BeforeGetStatus((inner, request) =>
+                        {
+                            cts.Cancel();
+                            scriptStatusRequest = request;
+                        })
+                        .BeforeCancelScript(() =>
+                        {
+                            if (scriptServiceExceptions.CancelScriptLatestException == null)
+                            {
+                                responseMessageTcpKiller.KillConnectionOnNextResponse();
+                            }
+                        })
+                        .Build())
+                    .Build())
+                .Build(CancellationToken);
+
+            var waitForFile = Path.Combine(clientTentacle.TemporaryDirectory.DirectoryPath, "waitforme");
+
+            var startScriptCommand = new StartScriptCommandV2Builder()
+                .WithScriptBody(new ScriptBuilder()
+                    .Print("hello")
+                    .WaitForFileToExist(waitForFile)
+                    .Print("AllDone"))
+                .Build();
+
+            List<ProcessOutput> logs = new List<ProcessOutput>();
+
+            Assert.ThrowsAsync<HalibutClientException>(async () => await clientTentacle.TentacleClient.ExecuteScriptAssumingException(startScriptCommand, logs, cts.Token));
+
+            // Let the script finish.
+            File.WriteAllText(waitForFile, "");
+            var legacyTentacleClient = clientTentacle.LegacyTentacleClientBuilder().Build(CancellationToken);
+            await Wait.For(() => legacyTentacleClient.ScriptService.GetStatus(new ScriptStatusRequest(scriptStatusRequest.Ticket, scriptStatusRequest.LastLogSequence)).State == ProcessState.Complete, CancellationToken);
+
+            var allLogs = logs.JoinLogs();
+
+            allLogs.Should().NotContain("AllDone");
+            scriptServiceExceptions.CancelScriptLatestException.Should().NotBeNull();
+            scriptServiceCallCounts.StartScriptCallCountStarted.Should().Be(1);
+            scriptServiceCallCounts.CancelScriptCallCountStarted.Should().Be(1);
+            scriptServiceCallCounts.CompleteScriptCallCountStarted.Should().Be(0);
```

Comment:
I think this is a bug. When we Cancel Script Execution we should also try and clean-up